src/main/java/com/cloudbees/jenkins/plugins/bitbucket/credentials/BitbucketOAuthCredentialsImpl.java

```diff
@@ -0,0 +1,41 @@
+package com.cloudbees.jenkins.plugins.bitbucket.credentials;
+
+
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import com.cloudbees.plugins.credentials.impl.BaseStandardCredentials;
+import hudson.Extension;
+import hudson.util.Secret;
+import javax.annotation.CheckForNull;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+
+public class BitbucketOAuthCredentialsImpl extends BaseStandardCredentials implements BitbucketOAuthCredentials {
+
+    private final String key;
+    private final Secret secret;
+    @DataBoundConstructor
+    public BitbucketOAuthCredentialsImpl(@CheckForNull CredentialsScope scope, @CheckForNull String id, String key,
+            String secret, @CheckForNull String description) {
+        super(scope, id, description);
+        this.key = key;
+        this.secret = Secret.fromString(secret);
```

Comment:
```suggestion
    public BitbucketOAuthCredentialsImpl(@CheckForNull CredentialsScope scope, @CheckForNull String id, String key,
            Secret secret, @CheckForNull String description) {
        super(scope, id, description);
        this.key = key;
        this.secret = secret;
```

Reply:
The guide about credentials https://github.com/jenkinsci/credentials-plugin/blob/master/docs/implementation.adoc#walk-through suggest to use String

Reply:
Than the guide is outdated and all plugins following that guide are a security risk.

Reply:
The guide seems to suggest the practice due to testing, which seems ill adviced.